plugins/package-managers/bower/src/main/kotlin/Bower.kt

```diff
@@ -183,14 +183,15 @@ private fun hasCompleteDependencies(info: PackageInfo, scopeName: String): Boole
     return dependencyKeys.containsAll(dependencyRefKeys)
 }
 
-private fun dependencyKeyOf(info: PackageInfo): String? {
-    // As non-null dependency keys are supposed to define an equivalence relation for parsing 'missing' nodes,
-    // only the name and version attributes can be used. Typically, those attributes should be not null
-    // however in particular for root projects the null case also happens.
-    val name = info.pkgMeta.name.orEmpty()
-    val version = info.pkgMeta.version.orEmpty()
-    return "$name:$version".takeUnless { name.isEmpty() || version.isEmpty() }
-}
+private val PackageInfo.key: String?
+    get() = with(pkgMeta) {
+        // As non-null dependency keys are supposed to define an equivalence relation for parsing 'missing' nodes,
+        // only the name and version attributes can be used. Typically, those attributes should be not null
+        // however in particular for root projects the null case also happens.
+        val name = pkgMeta.name.orEmpty()
+        val version = pkgMeta.version.orEmpty()
+        "$name:$version".takeUnless { name.isEmpty() || version.isEmpty() }
```

Comment:
As you have `with(pkgMeta)` above, I believe this could be simplified to just

    "$name:$version".takeUnless { name == null || version == null }

Reply:
Would this be a change in behavior in case of of them is `""` ?

Reply:
Anyhow, good catch with the `with(pkgMeta)`. This was a left over which I dropped now.

Reply:
> Would this be a change in behavior in case of of them is `""` ?

Yes, empty names / versions would not be discarded anymore. Not sure if this would actually be any issue though.